### PR TITLE
Signup: Add plan specific signup flows for 3Y plans

### PR DESCRIPTION
**Pull request by claudiucelfilip**
Related to #74456

## Proposed Changes

* This adds signup flows for the 3Y Starter, Explorer, Creator and Entrepreneur plans.

## Testing Instructions

* Test the new plan flows:
  * /start/personal-3y
  * /start/premium-3y
  * /start/business-3y
  * /start/ecommerce-3y
* The flows should skip the plans step
* Checkout should contain the correct plan, with the 3-year duration

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/signup/config/flows-pure.js [hunk 1]
```diff
@@ -573,6 +573,15 @@ export function generateFlows( {
 			showRecaptcha: true,
 			hideProgressIndicator: true,
 		},
+		{
+			name: 'ecommerce-3y',
```
Comment (niranjan-uma-shankar) [inline client/signup/config/flows-pure.js:577]: I think this flow is not used anywhere?

**Review by niranjan-uma-shankar — APPROVED** [review]
LGTM.

I left a comment. The eCommerce plan on the pricing grid uses the Stepper framework, so you'd need to implement the 3Y plan over there.

File: client/signup/config/flows-pure.js (hunk 1 above)
Comment (claudiucelfilip) [reply client/signup/config/flows-pure.js:577]: Probably not, but it appears to work and there's an `ecommerce-2y` so I decided to keep things consistent.

**Comment by claudiucelfilip** [comment]
> LGTM.
> 
> I left a comment. The eCommerce plan on the pricing grid uses the Stepper framework, so you'd need to implement the 3Y plan over there.

Thanks for the review!
The 3Y should been implemented already in https://github.com/Automattic/wp-calypso/pull/74456.

**Comment by niranjan-uma-shankar** [comment]
Ah okay, tested it and Stepper 3Y works too 👍

File: client/signup/config/flows-pure.js (hunk 1 above)
Comment (niranjan-uma-shankar) [reply client/signup/config/flows-pure.js:577]: 👍

**Pull request merged**